#### Open and read the file `Hawking-Questioning-the-Universe.txt`.

In [137]:
with open("Hawking-Questioning-the-Universe.txt", "r", encoding="utf-8") as f:
    text = f.read()
   # text = ''.join(text)

#### Print the first 50 characters.

In [119]:
print(text[:50])

There is nothing bigger or older than the universe


#### Print from the 52nd to the 163rd character.

In [120]:
print(text[51:163])

 The questions I would like to talk about are: one, where did we come from? How did the universe come into being


---

#### Open the file `Hawking-Questioning-the-Universe.txt` and create a list named `text_as_list` where every word is an element of the list. 

Hint: there are two ways to do it. Either with `.readlines()` or with `.split()`. I encourage you to google how both work, but I suggest to adopt the latter.

Hint: if you want to split the text so every word is an element of a list, what should be the parameter of the `split()` method? Read more [here](https://www.w3schools.com/python/ref_string_split.asp).

In [135]:
with open("Hawking-Questioning-the-Universe.txt", "r", encoding="utf-8") as f:
    text_as_list = f.read().split(' ')

#### This code will remove the punctuation from your list and applies the lower() method that returns a string where all characters are lower case. Simple run the cell.

In [134]:
import string
text_as_list = [i.translate(str.maketrans('', '', string.punctuation)).lower() for i in text_as_list]

#### Create a new list `words_begin_with_a`. Write a for loop that checks if a word starts with the letter "a". If yes, append the word to the list `words_begin_with_a`. Print how many words have been appended to the list.

Hint: you have to define first an empty list.

Hint: in order to find if a string starts with one or more letters, you should use the method `.startswith()`. Read more [here](https://www.w3schools.com/python/ref_string_startswith.asp).

In [138]:
words_begin_with_a = []
for word in text:
    if word.startswith('a'):
        words_begin_with_a.append(word)
print(len(words_begin_with_a))

310


Answer: 310

#### Write an if statement that checkes if the word "universe" appears in the text. Feel free to google how we test if a word is _in_ a Python list.

In [92]:
#if text_as_list.index('universe'): #this is my pre-Google method
    #print("this word appears")
if "universe" in text_as_list: 
    print("this word appears")

this word appears


#### Write an if statement that checkes if the word "alliens" appears in the text. If not print "the word does not appear"

In [91]:
if "alliens" in text_as_list: 
    print("this word appears")
else: 
    print("the word does not appear")

the word does not appear


#### Write an if statement that checkes if the word "alliens" appears in the text. If not, check if the word "International Space Station" appears in the text. If neither is in the text, print "I didn't find the words".

In [94]:
if "alliens" in text_as_list:
    print("alliens appear")
elif "International Space Station" in text_as_list:
    print("International Space Station appears")
else: 
    print("I didn't find these words")

I didn't find these words


#### Write a function that prints the numbers from 1 to 20. But for multiples of three, print "Fizz" instead of the number, and for the multiples of five, print "Buzz." For numbers that are multiples of both three and five, print "FizzBuzz."

In [106]:
for x in range(1,21):
    if x % 3 == 0 and x % 5 == 0:
        print("FizzBuzz")
    elif x % 3 == 0:
        print("Fizz")
    elif x % 5 == 0:
        print("Buzz")
    else:
        print(x)

1
2
Fizz
4
Buzz
Fizz
7
8
Fizz
Buzz
11
Fizz
13
14
FizzBuzz
16
17
Fizz
19
Buzz


#### Password Strength Checker:
Write a function that takes a password as input and checks its strength based on the following criteria:

- At least 8 characters long
- Contains both uppercase and lowercase letters
- Contains at least one digit

Hint: use the string methods `.isupper()`, `.islower()` and `.isdigit()`

In [159]:
def password_checker(password): 
    if len(password) < 8:
        return False 
    x = "str"
    if any(x.isupper for x in password) and any(x.islower for x in password):
        return True
    else: 
        return False

False


#### Leap Year Checker
Write a function that takes a year as input and returns whether it's a leap year or not.

In [105]:
# https://nrc.canada.ca/en/certifications-evaluations-standards/canadas-official-time/what-years-are-leap-years
def leap_year_checker(year):
    if year % 400 == 0:
        return True
    elif year % 100 == 0:
        return False
    elif year % 4 == 0:
        return True
    else: 
        return False

True
